samcli/local/docker/lambda_container.py

```diff
@@ -87,12 +90,51 @@ def _get_exposed_ports(debug_port):
         :param int debug_port: Optional, integer value of debug port
         :return dict: Dictionary containing port binding information. None, if debug_port was not given
         """
-        if not debug_port:
+        if not debug_options:
             return None
 
         return {
             # container port : host port
-            debug_port: debug_port
+            debug_options.debug_port: debug_options.debug_port
+        }
+
+    @staticmethod
+    def _get_additional_options(runtime, debug_options):
+        """
+        Return additional Docker container options. Used by container debug mode to enable certain container
+        security options.
+        :param DebugContext debug_options: DebugContext for the runtime of the container.
+        :return dict: Dictionary containing additional arguments to be passed to container creation.
+        """
+        if not debug_options:
+            return None
+
+        opts = {}
+
+        if runtime == Runtime.go1x.value:
+            # These options are required for delve to function properly inside a docker container on docker < 1.12
+            # See https://github.com/moby/moby/issues/21051
+            opts["security_opt"] = ["seccomp:unconfined"]
+            opts["cap_add"] = ["SYS_PTRACE"]
+
+        return opts
+
+    @staticmethod
+    def _get_additional_volumes(debug_options):
+        """
+        Return additional volumes to be mounted in the Docker container. Used by container debug for mapping
+        debugger executable into the container.
+        :param DebugContext debug_options: DebugContext for the runtime of the container.
+        :return dict: Dictionary containing volume map passed to container creation.
+        """
+        if not debug_options or not debug_options.debugger_path:
+            return None
+
+        return {
+            debug_options.debugger_path: {
+                "bind": "/tmp/lambci_debug_files",
```

Comment:
Can you move this static string to a class member? Also add comments explaining if this is the path in local file system or inside docker container

Reply:
Good call!